cmd/createorupdate/createorupdate.go

```diff
@@ -202,9 +202,19 @@ func main() {
 		log.Fatal(err)
 	}
 
+	//simualte Context with property bag
+	pb := make(map[string]string)
+	pb["ClientID"] = os.Getenv("AZURE_CLIENT_ID")
```

Comment:
any reason not to do:

in api.go:

```
type ContextKey string

const (
	ContextKeyClientID ContextKey = "ClientID"
	ContextKeyClientSecret ContextKey = "ClientSecret"
	...
)
```

here:

```
ctx := context.Background()
ctx = ctx.WithValue(ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID")
ctx = ctx.WithValue(ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET")
...
```

at point of use:

```
config := auth.NewClientCredentialsConfig(ctx.Value(ContextKeyClientID).(string), ...)
```

Avoids v1.PropertyBagKey and the additional map.

Reply:
like it :)